export unitsplit

orichters opened the pull request:
- put it into `unitsplit.R`, slightly adapted such that it also accepts strings or vectors of strings: if `ncol` of `x` is `NULL`, it simply puts it into a data.frame. `unitsplit()` can now be called not with column number, but name.
- also separate the tests
- add Mika and myself as authors

mikapfl reviewed (approved):
Nice, thanks! Maybe add a test which just uses a vector of strings instead of a data frame, to make sure we don't break this functionality again later?

mikapfl commented:
Also, I'm not sure about the pre-commit.ci failures, you didn't change anything in the dependencies, so I guess it at least doesn't get worse with this PR.

mikapfl commented:
@johanneskoch94 you added https://github.com/pik-piam/magclass/blob/4df6a63c36df685bc1a95db6017875e5d151927c/R/as_tibble.R#L11 , but didn't add a dependency on pkgconfig in the DESCRIPTION. Do you know how to best resolve this (not related to this PR, just the place where it crops up due to better linting).

orichters commented:
> @johanneskoch94 you added
> 
> https://github.com/pik-piam/magclass/blob/4df6a63c36df685bc1a95db6017875e5d151927c/R/as_tibble.R#L11
> , but didn't add a dependency on pkgconfig in the DESCRIPTION. Do you know how to best resolve this (not related to this PR, just the place where it crops up due to better linting).

added them to `suggested` and pre-commit is fine now. Also added further tests and the functionality to call the `col` by name, not only number.

tscheypidi commented on DESCRIPTION:
```diff
@@ -45,6 +47,8 @@ Suggests:
     knitr,
     ncdf4,
+    pkgconfig,
+    quitte,
```
I think adding `quitte` here creates problems when updating `magclass` on CRAN as `quitte` is not available there. Can you remove it again from the `suggests` list or is it required?

orichters replied:
I added it because pre-commit.ci was complaining…

tscheypidi replied:
ah ok. I thought it was only complaining about pkgconfig. If both was needed, then we can keep it for the moment and I will find out with my next CRAN-update whether this is accepted there or not.

orichters replied:
See https://results.pre-commit.ci/run/github/93050064/1666187376.mH43gdB9QXCEJrhPgUS4Jw
```
Not all packages used in your code are listed in DESCRIPTION. The following are missing in file `/code/tests/testthat/test-as.magpie.R`: quitte.
```
On the other hand, I'm not sure what were just warnings and what lead to failure.

mikapfl replied:
strictly speaking, quitte is not a dependency, because its use is guarded by checks if it is installed. So, technically, we don't need any kind of dependency, but a suggests also doesn't feel wrong.

orichters replied:
"Packages on which a CRAN package depends should be available from a mainstream repository: if any mentioned in ‘Suggests’ or ‘Enhances’ fields are not from such a repository, where to obtain them at a repository should be specified in an ‘Additional_repositories’ field of the DESCRIPTION file (as a comma-separated list of repository URLs) or for other means of access, described in the ‘Description’ field." ([source](https://cran.r-project.org/web/packages/policies.html)).

Maybe this is an option? Or submit `quitte` to CRAN?

tscheypidi replied:
ah cool....so Additional_repositories might then in fact be a feasible solution.

@mikapfl There is no doubt that it makes sense to mention it, but it was more the issue with the checks which come when you upload it to CRAN.
